apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/create/components/client/create-ecosystem-form.client.tsx

```diff
@@ -0,0 +1,192 @@
+              render={() => (
+                <FormItem>
+                  <FormLabel>Ecosystem Logo</FormLabel>
+                  <FormControl>
+                    <ImageUpload
+                      accept="image/png, image/jpeg"
+                      onUpload={(files) => {
+                        form.setValue("logo", files[0]);
+                        form.clearErrors("logo");
+                      }}
+                    />
+                  </FormControl>
+                  <FormDescription>
+                    The logo will be displayed as the ecosystem's wallet icon.
+                  </FormDescription>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+            <FormField
+              control={form.control}
+              name="permission"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>Integration permissions</FormLabel>
+                  <FormControl>
+                    <RadioGroup
+                      onValueChange={field.onChange}
+                      defaultValue="PARTNER_WHITELIST"
+                      className="flex gap-4 py-2"
+                    >
+                      <FormItem>
+                        <FormControl>
+                          <RadioGroupItemButton
+                            value="PARTNER_WHITELIST"
+                            id="PARTNER_WHITELIST"
+                          >
+                            Allowlist
+                          </RadioGroupItemButton>
+                        </FormControl>
+                      </FormItem>
+                      <FormItem>
+                        <FormControl>
+                          <RadioGroupItemButton value="ANYONE" id="ANYONE">
+                            Public
+                          </RadioGroupItemButton>
+                        </FormControl>
+                      </FormItem>
+                    </RadioGroup>
+                  </FormControl>
+                  <FormDescription>
+                    <Link
+                      className="text-primary"
+                      href="https://portal.thirdweb.com/connect/ecosystems/ecosystem-permissions"
+                    >
+                      Learn more
+                    </Link>{" "}
+                    about ecosystem permissions
+                  </FormDescription>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          </div>
+          <Button
+            type="submit"
+            variant="primary"
+            className="min-w-28"
+            disabled={isLoading}
+          >
+            {isLoading && <Loader2 className="w-4 h-4 mr-2 animate-spin" />}
+            Create
+          </Button>
+        </form>
+      </Form>
+      <ConfirmationDialog
+        open={formDataToBeConfirmed !== undefined}
+        onOpenChange={(open) =>
+          !open ? setFormDataToBeConfirmed(undefined) : null
```

Comment:
this feels odd, should be able to just do this after the `onSubmit` finishes, no?